Do not substitute parameters during SQL/Expression formatting

martint opened the pull request:
This is the wrong place to do so. The formatter should do a literal
translation from the syntax tree to the textual representation.

findepi commented on presto-main/src/main/java/io/prestosql/execution/DataDefinitionTask.java:
```diff
@@ -33,11 +32,21 @@
+        if (!parameters.isEmpty()) {
+            String formatted = parameters.stream()
+                    .map(Expression::toString)
+                    .collect(Collectors.joining(", ", "[", "]"));
```
formatted eq parameters.toString()

(List#toString is documented, stable, AFAIK)

in fact, you don't use `formatted` at all

(also, static import .joining)

martint replied:
Left-over from some work-in-progress...

findepi commented on presto-main/src/test/java/io/prestosql/security/TestFileBasedSystemAccessControl.java:
```diff
@@ -270,8 +269,8 @@ public void parseUnknownRules()
+    private void parse(String path)
     {
-        return new FileBasedSystemAccessControl.Factory().create(ImmutableMap.of(SECURITY_CONFIG_FILE, path));
+        new FileBasedSystemAccessControl.Factory().create(ImmutableMap.of(SECURITY_CONFIG_FILE, path));
```
unrelated change

findepi commented on presto-main/src/test/java/io/prestosql/security/TestFileBasedSystemAccessControl.java:
```diff
@@ -104,7 +104,6 @@ public void testCanSetUserOperations()
         catch (AccessDeniedException expected) {
         }
 
-        TransactionManager transactionManagerNoPatterns = createTestTransactionManager();
```
unrelated change

martint replied:
Weird, no idea how these happened.

martint committed:
Do not substitute parameters during SQL/Expression formatting

This is the wrong place to do so. The formatter should do a literal
translation from the syntax tree to the textual representation.
